File: fixtures/scale/python/large/cipherlab/checks/check_staticpad.py
from cipherlab.ciphers.staticpad import staticpad_encrypt, staticpad_decrypt
from cipherlab.specs.staticpad_spec import staticpad_spec
from cipherlab.corpus.corpus_mask import corpus_mask


def check_staticpad():
    spec = staticpad_spec()
    for text in corpus_mask():
        sealed = staticpad_encrypt(text, spec["key"])
        opened = staticpad_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "mask"

File: fixtures/scale/ruby/large/lib/check_staticpad.rb
require_relative 'staticpad_encrypt'
require_relative 'staticpad_decrypt'
require_relative 'corpus_mask'

def check_staticpad
  corpus_mask.each do |text|
    sealed = staticpad_encrypt(text, 7)
    opened = staticpad_decrypt(sealed, 7)
    return false if opened != text
  end
  true
end
